### PR TITLE
⬆️ Upgrade Web Server service (Pydantic v2)

### PR DESCRIPTION
**Pull request by giancarloromeo**
## What do these changes do?
Upgrade the Web Server service to Pydantic v2.

## How to test

```shell
make devenv
source .venv/bin/activate
cd services/web/server
make install-dev
make mypy
make test-dev-unit
```

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: packages/pytest-simcore/src/pytest_simcore/hypothesis_type_strategies.py [hunk 1]
```diff
@@ -1,9 +1,5 @@
-from hypothesis import provisional
-from hypothesis import strategies as st
-from pydantic import AnyHttpUrl, AnyUrl, HttpUrl
-
 # FIXME: For now it seems the pydantic hypothesis plugin does not provide strategies for these types.
 # therefore we currently provide it
-st.register_type_strategy(AnyUrl, provisional.urls())
-st.register_type_strategy(HttpUrl, provisional.urls())
-st.register_type_strategy(AnyHttpUrl, provisional.urls())
+# st.register_type_strategy(AnyUrl, provisional.urls())
```
Comment (giancarloromeo) [inline packages/pytest-simcore/src/pytest_simcore/hypothesis_type_strategies.py:3]: Unsupported by Hypothesis, a solution must be found.

File: packages/common-library/src/common_library/pydantic_validators.py [hunk 2]
```diff
@@ -1,10 +1,35 @@
...
+def _validate_legacy_timedelta_str(time_str: str | timedelta) -> str | timedelta:
+    if not isinstance(time_str, str):
+        return time_str
+
+    # Match the format [-][DD ][HH:MM]SS[.ffffff]
+    match = re.match(
+        r"^(?P<sign>-)?(?:(?P<days>\d+)\s)?(?:(?P<hours>\d+):)?(?:(?P<minutes>\d+):)?(?P<seconds>\d+)(?P<fraction>\.\d+)?$",
+        time_str,
+    )
+    if not match:
```
Comment (sanderegg) [inline packages/common-library/src/common_library/pydantic_validators.py:18]: why is this one needed?
pydantic handles this format for date times: YYYY-MM-DD[T]HH:MM[:SS[.ffffff]][Z or [±]HH[:]MM], [-][DD ][HH:MM]SS[.ffffff]
[-][DD ][HH:MM]SS[.ffffff], [±]P[DD]DT[HH]H[MM]M[SS]S (ISO 8601 format for timedelta)

File: packages/models-library/src/models_library/api_schemas_directorv2/clusters.py [hunk 3]
```diff
@@ -96,7 +97,7 @@ class ClusterGet(Cluster):
         alias="accessRights", default_factory=dict
     )
 
-    model_config = ConfigDict(populate_by_name=True)
+    model_config = ConfigDict(extra="allow", populate_by_name=True)
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_directorv2/clusters.py:100]: why is this change needed? I can check it, maybe if you add my name or please copy it for me in the main issue

File: packages/models-library/src/models_library/api_schemas_directorv2/clusters.py [hunk 4]
```diff
@@ -167,15 +168,15 @@ def set_default_thumbnail_if_empty(cls, v, values):
...
+    name: str | None = None  # type: ignore[assignment]
+    description: str | None = None
+    type: ClusterTypeInModel | None = None  # type: ignore[assignment]
+    owner: GroupID | None = None  # type: ignore[assignment]
+    thumbnail: HttpUrl | None = None
+    endpoint: AnyUrl | None = None  # type: ignore[assignment]
+    authentication: ExternalClusterAuthentication | None = None  # type: ignore[assignment]
     access_rights: dict[GroupID, ClusterAccessRights] | None = Field(  # type: ignore[assignment]
-        alias="accessRights"
+        None, alias="accessRights"
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_directorv2/clusters.py:179]:
```suggestion
        default=None, alias="accessRights"
```

File: packages/models-library/src/models_library/api_schemas_webserver/auth.py [hunk 5]
```diff
@@ -83,9 +83,10 @@ class ApiKeyGet(BaseModel):
     api_secret: str
 
     model_config = ConfigDict(
+        from_attributes=True,
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_webserver/auth.py:86]: is this used in conjonction with the DB? if yes how was it even working before?

File: packages/models-library/src/models_library/api_schemas_webserver/clusters.py [hunk 6]
```diff
@@ -8,7 +8,7 @@
 class ClusterPathParams(BaseModel):
     cluster_id: ClusterID
     model_config = ConfigDict(
-        populate_by_name=True, 
+        populate_by_name=True,
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_webserver/clusters.py:11]: not sure I see the point of this change? there is no alias so what does it do?

File: packages/models-library/src/models_library/api_schemas_webserver/folders_v2.py [hunk 7]
```diff
@@ -45,7 +45,7 @@ class CreateFolderBodyParams(InputSchema):
 
 class PutFolderBodyParams(InputSchema):
     name: IDStr
-    parent_folder_id: FolderID | None
+    parent_folder_id: FolderID | None = None
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_webserver/folders_v2.py:48]: this is a change in syntax. @matusdrobuliak66?

File: packages/models-library/src/models_library/api_schemas_webserver/groups.py [hunk 8]
```diff
@@ -90,7 +91,7 @@ def _sanitize_legacy_data(cls, v):
         if v:
             # Enforces null if thumbnail is not valid URL or empty
             with suppress(ValidationError):
-                return TypeAdapter(AnyUrl).validate_python(v)
+                return TypeAdapter(AnyHttpUrl).validate_python(v)
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_webserver/groups.py:94]: is this really the wanted change? going from Any url to http?

File: packages/models-library/src/models_library/api_schemas_webserver/projects.py [hunk 9]
```diff
@@ -105,7 +107,7 @@ class ProjectReplace(InputSchema):
     uuid: ProjectID
     name: ShortTruncatedStr
     description: LongTruncatedStr
-    thumbnail: HttpUrl | None
+    thumbnail: Annotated[HttpUrl | None, BeforeValidator(empty_str_to_none_pre_validator)] = Field(default=None)
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_webserver/projects.py:110]: nice!

File: packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py [hunk 10]
```diff
@@ -19,7 +19,7 @@
 class NodeCreate(InputSchemaWithoutCamelCase):
     service_key: ServiceKey
     service_version: ServiceVersion
-    service_id: str | None
+    service_id: str | None = Field(default=None)
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py:22]: this is also a change in syntax.

File: packages/models-library/src/models_library/api_schemas_webserver/resource_usage.py [hunk 11]
```diff
@@ -131,7 +131,7 @@ class UpdatePricingUnitBodyParams(InputSchema):
     unit_extra_info: UnitExtraInfo
     default: bool
     specific_info: SpecificInfo
-    pricing_unit_cost_update: PricingUnitCostUpdate | None
+    pricing_unit_cost_update: PricingUnitCostUpdate | None = Field(default=None)
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_webserver/resource_usage.py:134]: and here as well

File: packages/models-library/src/models_library/api_schemas_webserver/wallets.py [hunk 12]
```diff
@@ -20,7 +20,10 @@ class WalletGet(OutputSchema):
     created: datetime
     modified: datetime
 
-    model_config = ConfigDict(frozen=False)
+    model_config = ConfigDict(
+        from_attributes=True,
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_webserver/wallets.py:24]: why this change?

File: packages/models-library/src/models_library/clusters.py [hunk 13]
```diff
@@ -159,6 +159,7 @@ class Cluster(BaseCluster):
     id: ClusterID = Field(..., description="The cluster ID")
 
     model_config = ConfigDict(
+        extra="allow",
```
Comment (sanderegg) [inline packages/models-library/src/models_library/clusters.py:162]: why is this necessary?

File: packages/models-library/src/models_library/resource_tracker.py [hunk 14]
```diff
@@ -259,7 +259,7 @@ class PricingUnitWithCostUpdate(BaseModel):
     unit_extra_info: UnitExtraInfo
     default: bool
     specific_info: SpecificInfo
-    pricing_unit_cost_update: None | PricingUnitCostUpdate
+    pricing_unit_cost_update: None | PricingUnitCostUpdate = Field(default=None)
```
Comment (sanderegg) [inline packages/models-library/src/models_library/resource_tracker.py:262]: also a change. if these changes you do because of tests. I think the problem lies with the tests and we should not change the interface

File: packages/models-library/src/models_library/rest_pagination_utils.py [hunk 15]
```diff
@@ -69,9 +70,11 @@ def paginate_data(
     """
     last_page = ceil(total / limit) - 1
 
+    data = [item.model_dump() if hasattr(item, "model_dump") else item for item in chunk]
+
     return PageDict(
         _meta=PageMetaInfoLimitOffset(
-            total=total, count=len(chunk), limit=limit, offset=offset
+            total=total, count=len(data), limit=limit, offset=offset
         ),
```
Comment (sanderegg) [inline packages/models-library/src/models_library/rest_pagination_utils.py:78]: not sure I understand that change. chunk is a list, so why does it need a change?

File: packages/pytest-simcore/src/pytest_simcore/hypothesis_type_strategies.py [hunk 16]
```diff
@@ -1,9 +1,13 @@
...
-# therefore we currently provide it
-st.register_type_strategy(AnyUrl, provisional.urls())
-st.register_type_strategy(HttpUrl, provisional.urls())
-st.register_type_strategy(AnyHttpUrl, provisional.urls())
+
+@composite
+def url_strategy(draw):
+    return TypeAdapter(Url).validate_python(draw(provisional.urls()))
+
+
```
Comment (sanderegg) [inline packages/pytest-simcore/src/pytest_simcore/hypothesis_type_strategies.py:12]: so does it mean that works now like so?

File: services/web/server/src/simcore_service_webserver/api_keys/_db.py [hunk 17]
```diff
@@ -79,7 +79,7 @@ async def get(
 
             result: ResultProxy = await conn.execute(stmt)
             row: RowProxy | None = await result.fetchone()
-            return ApiKeyInDB.from_orm(row) if row else None
+            return ApiKeyInDB.model_validate(row) if row else None
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/api_keys/_db.py:82]: so ApiKeyInDB needs to have the from_attribute set to True, so why did you change it on ApiKey model?

**Review by sanderegg — APPROVED** [review]
great job! thanks!

File: packages/models-library/src/models_library/api_schemas_directorv2/clusters.py (hunk 3 above)
Comment (giancarloromeo) [reply packages/models-library/src/models_library/api_schemas_directorv2/clusters.py:100]: If you remove it, a test fails because the `"forbid"` behavior is inherited from `BaseCluster` class.

File: packages/models-library/src/models_library/clusters.py (hunk 13 above)
Comment (giancarloromeo) [reply packages/models-library/src/models_library/clusters.py:162]: https://github.com/ITISFoundation/osparc-simcore/pull/6584#discussion_r1828945148

File: packages/models-library/src/models_library/resource_tracker.py (hunk 14 above)
Comment (giancarloromeo) [reply packages/models-library/src/models_library/resource_tracker.py:262]: So, the test has been written afterwards?! What's the purpose of the tests then??

File: packages/common-library/src/common_library/pydantic_validators.py (hunk 2 above)
Comment (giancarloromeo) [reply packages/common-library/src/common_library/pydantic_validators.py:18]: This is not supported. That's why I would also remove all these "legacy" solutions and migrate all the _intervals_ to appropriate standard format.

https://github.com/giancarloromeo/osparc-simcore/blob/ded3b252c6012574739ddc30f480147e1f3468f0/services/web/server/tests/unit/isolated/test_studies_dispatcher_settings.py#L33

File: packages/models-library/src/models_library/api_schemas_webserver/projects.py (hunk 9 above)
Comment (giancarloromeo) [reply packages/models-library/src/models_library/api_schemas_webserver/projects.py:110]: All the "external validators" could disappear and write this compact way in these situations.

Comment (matusdrobuliak66) [reply packages/models-library/src/models_library/api_schemas_webserver/projects.py:110]: cool :+1:

File: packages/pytest-simcore/src/pytest_simcore/hypothesis_type_strategies.py (hunk 16 above)
Comment (giancarloromeo) [reply packages/pytest-simcore/src/pytest_simcore/hypothesis_type_strategies.py:12]: In the past Url was a `str`, so the provisional.urls() created the right types for the strategy. Now it's a `Url` and must be properly instantiated before "injecting it".

**Review by GitHK — COMMENTED** [review]
I have some concerns that might cause issues. Please see them below.

File: packages/models-library/src/models_library/api_schemas_directorv2/clusters.py [hunk 18]
```diff
@@ -167,15 +168,15 @@ def set_default_thumbnail_if_empty(cls, v, values):
...
+    name: str | None = None  # type: ignore[assignment]
+    description: str | None = None
+    type: ClusterTypeInModel | None = None  # type: ignore[assignment]
+    owner: GroupID | None = None  # type: ignore[assignment]
+    thumbnail: HttpUrl | None = None
+    endpoint: AnyUrl | None = None  # type: ignore[assignment]
+    authentication: ExternalClusterAuthentication | None = None  # type: ignore[assignment]
     access_rights: dict[GroupID, ClusterAccessRights] | None = Field(  # type: ignore[assignment]
-        alias="accessRights"
+        default=None, alias="accessRights"
```
Comment (GitHK) [inline packages/models-library/src/models_library/api_schemas_directorv2/clusters.py:179]: why have all these become None by default?

File: packages/common-library/src/common_library/pydantic_validators.py [hunk 19]
```diff
@@ -1,10 +1,35 @@
...
+    hours = match.group("hours") or "0"
+    minutes = match.group("minutes") or "0"
+    seconds = match.group("seconds")
+    fraction = match.group("fraction") or ""
+
+    # Convert to the format [-][DD]D[,][HH:MM:]SS[.ffffff]
+    return f"{sign}{int(days)}D,{int(hours):02}:{int(minutes):02}:{seconds}{fraction}"
+
+
 def validate_numeric_string_as_timedelta(field: str):
```
[the remark this answers is not in the document]
Comment (giancarloromeo) [reply packages/common-library/src/common_library/pydantic_validators.py:33]: I agree, see the other comment written above: https://github.com/ITISFoundation/osparc-simcore/pull/6584#discussion_r1828964392

File: packages/models-library/src/models_library/api_schemas_webserver/resource_usage.py (hunk 11 above)
Comment (matusdrobuliak66) [reply packages/models-library/src/models_library/api_schemas_webserver/resource_usage.py:134]: Here its fine to leave default value None. (also FYI this endpoint is used just internally by us "admin" users)

File: packages/models-library/src/models_library/api_schemas_webserver/folders_v2.py (hunk 7 above)
Comment (matusdrobuliak66) [reply packages/models-library/src/models_library/api_schemas_webserver/folders_v2.py:48]: Here, it's fine to have the default value as None. (We could also make it required since the frontend currently always fills this field.) However, we can go with the easier solution; both options are acceptable.

File: packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py (hunk 10 above)
Comment (matusdrobuliak66) [reply packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py:22]: @pcrespov ?

**Review by matusdrobuliak66 — APPROVED** [review]
Thanks!!

File: services/web/server/src/simcore_service_webserver/projects/_workspaces_handlers.py [hunk 20]
```diff
@@ -50,14 +50,12 @@ async def wrapper(request: web.Request) -> web.StreamResponse:
 
 class _ProjectWorkspacesPathParams(BaseModel):
     project_id: ProjectID
-    workspace_id: WorkspaceID | None
-
-    class Config:
-        extra = Extra.forbid
+    workspace_id: WorkspaceID | None = None
```
[the remark this answers is not in the document]
Comment (giancarloromeo) [reply services/web/server/src/simcore_service_webserver/projects/_workspaces_handlers.py:53]: Asked @matusdrobuliak66

File: packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py (hunk 10 above)
Comment (giancarloromeo) [reply packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py:22]: OpenAPI Spec says it's not required. I consider any failing test as error in the test itself

Comment (giancarloromeo) [reply packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py:22]: OpenAPI Spec (3.0.1) - master branch:

```yaml
NodeCreate:
      title: NodeCreate
      required:
      - service_key
      - service_version
      type: object
      properties:
        service_key:
          title: Service Key
          pattern: ^simcore/services/((comp|dynamic|frontend))/([a-z0-9][a-z0-9_.-]*/)*([a-z0-9-_]+[a-z0-9])$
          type: string
        service_version:
          title: Service Version
          pattern: ^(0|[1-9]\d*)(\.(0|[1-9]\d*)){2}(-(0|[1-9]\d*|\d*[-a-zA-Z][-\da-zA-Z]*)(\.(0|[1-9]\d*|\d*[-a-zA-Z][-\da-zA-Z]*))*)?(\+[-\da-zA-Z]+(\.[-\da-zA-Z-]+)*)?$
          type: string
        service_id:
          title: Service Id
          type: string
```
I use this a source of truth, and consider any failing test as an error in the test itself.

File: packages/models-library/src/models_library/api_schemas_webserver/auth.py (hunk 5 above)
Comment (pcrespov) [reply packages/models-library/src/models_library/api_schemas_webserver/auth.py:86]: This model in particular needs quite some upgrades https://github.com/ITISFoundation/osparc-simcore/issues/4959

File: services/web/server/src/simcore_service_webserver/announcements/_models.py [hunk 21]
```diff
@@ -18,10 +18,10 @@ class Announcement(BaseModel):
     link: str
     widgets: list[Literal["login", "ribbon", "user-menu"]]
 
-    @validator("end")
+    @field_validator("end")
     @classmethod
-    def check_start_before_end(cls, v, values):
-        if start := values.get("start"):
+    def check_start_before_end(cls, v, info: ValidationInfo):
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/announcements/_models.py:23]: MINOR: I started protecting all access to validators since they should never be explicitly called. i.e. here I would rename as `_check_start_before_end` (i.e. with `_` prefix)

File: services/web/server/src/simcore_service_webserver/application_settings.py [hunk 22]
```diff
@@ -54,7 +63,7 @@ class ApplicationSettings(BaseCustomSettings, MixinLoggingSettings):
     # CODE STATICS ---------------------------------------------------------
     API_VERSION: str = API_VERSION
     APP_NAME: str = APP_NAME
-    API_VTAG: VersionTag = parse_obj_as(VersionTag, API_VTAG)
+    API_VTAG: VersionTag = TypeAdapter(VersionTag).validate_python(API_VTAG)
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/application_settings.py:66]: MINOR: in next PRs we can refactor all `ApplicationSettings` to inherit from `settings_library.application.BaseApplicationSettings`. That base class includes all env vars expected from the Dockerfile

File: services/web/server/src/simcore_service_webserver/application_settings.py [hunk 23]
```diff
@@ -250,57 +279,63 @@ class ApplicationSettings(BaseCustomSettings, MixinLoggingSettings):
         "Currently this is a system plugin and cannot be disabled",
     )
 
-    @root_validator()
+    model_config = SettingsConfigDict(
+        frozen=False
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/application_settings.py:283]: why not frozen? Is there a reason to modify the settings once they are constructed?

File: services/web/server/src/simcore_service_webserver/cli.py [hunk 24]
```diff
@@ -68,7 +68,8 @@ async def app_factory() -> web.Application:
     assert app_settings.SC_BUILD_TARGET  # nosec
 
     _logger.info(
-        "Application settings: %s", app_settings.json(indent=2, sort_keys=True)
+        "Application settings: %s",
+        app_settings.model_dump_json(indent=2),
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/cli.py:72]: MINOR: do I see we miss here the sort_keys ... What about using instead our own `json_dumps`, i.e.
```python
from models_library.utils.json_serialization import json_dumps

_logger.info( "Application settings: %s", json_dumps(app_settings, indent=2, sort_keys=True))
```

File: services/web/server/src/simcore_service_webserver/payments/_methods_api.py [hunk 25]
```diff
@@ -79,7 +79,7 @@ async def _fake_init_creation_of_wallet_payment_method(
     await asyncio.sleep(1)
     payment_method_id = PaymentMethodID(f"{_FAKE_PAYMENT_METHOD_ID_PREFIX}_{uuid4()}")
     form_link = (
-        URL(settings.PAYMENTS_FAKE_GATEWAY_URL)
+        URL(f"{settings.PAYMENTS_FAKE_GATEWAY_URL}")
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/payments/_methods_api.py:82]: THOUGHT: maybe we should have some helpers with converters from/to all URL classes we have. From the top of my head we are using at least these
- yarl.URL
- pydantic.HTTPUrl
- httpx.URL
- starlette.URL (via fastapi)

File: services/web/server/src/simcore_service_webserver/projects/_crud_api_read.py [hunk 26]
```diff
@@ -53,11 +53,11 @@ async def _append_fields(
     # replace project access rights (if project is in workspace)
     if workspace_access_rights:
         project["accessRights"] = {
-            gid: access.dict() for gid, access in workspace_access_rights.items()
+            f"{gid}": access.model_dump() for gid, access in workspace_access_rights.items()
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/projects/_crud_api_read.py:56]: it should be a int, right?

File: services/web/server/src/simcore_service_webserver/projects/_crud_handlers.py [hunk 27]
```diff
@@ -389,18 +389,13 @@ async def get_project_inactivity(request: web.Request):
     return web.json_response(Envelope(data=project_inactivity), dumps=json_dumps)
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/_crud_handlers.py:390]: Add the doc you removed inside `patch_project`  as a comment

```
# - Update https://google.aip.dev/134
```

Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/projects/_crud_handlers.py:390]: Same with the others ... in the middle, as they are now, they get misplaced with time .

File: services/web/server/src/simcore_service_webserver/projects/_nodes_api.py [hunk 28]
```diff
@@ -96,10 +95,10 @@ class NodeScreenshot(BaseModel):
     mimetype: str | None = Field(
         default=None,
         description="File's media type or None if unknown. SEE https://www.iana.org/assignments/media-types/media-types.xhtml",
-        example="image/jpeg",
+        examples=["image/jpeg"],
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/_nodes_api.py:98]: THOUGTH: we should extend the auto `tests__model__examples` to check these as well. For the moment we only check that the examples in the `model_config` validate against the pydantic class.

File: services/web/server/src/simcore_service_webserver/projects/_tags_api.py [hunk 29]
```diff
@@ -45,7 +45,8 @@ async def add_tag(
             )
         )
         project["accessRights"] = {
-            gid: access.dict() for gid, access in workspace_db.access_rights.items()
+            gid: access.model_dump()
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/_tags_api.py:48]: see?, here you do not convert `gid` to string? (ref. comment above)

File: services/web/server/src/simcore_service_webserver/security/_authz_db.py [hunk 30]
```diff
@@ -35,8 +35,12 @@ async def get_active_user_or_none(engine: Engine, email: str) -> AuthInfoDict |
             )
         )
         row = await result.fetchone()
-        assert row is None or parse_obj_as(IdInt, row.id) is not None  # nosec
-        assert row is None or parse_obj_as(UserRole, row.role) is not None  # nosec
+        assert (
+            row is None or TypeAdapter(IdInt).validate_python(row.id) is not None
+        )  # nosec
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/security/_authz_db.py:40]: NOTE: these `# nosec` must be **on the same line** as `assert`. 
For the moment, they do not have an effect but if we re-activate security checks it will fail.
Let's try to keep the correctly ... otherwise we will have to review them again

File: services/web/server/src/simcore_service_webserver/studies_dispatcher/_catalog.py [hunk 31]
```diff
@@ -97,7 +97,7 @@ async def iter_latest_product_services(
             )
             & (services_meta_data.c.deprecated.is_(None))
             & (services_access_rights.c.gid == EVERYONE_GROUP_ID)
-            & (services_access_rights.c.execute_access == True)
+            & (services_access_rights.c.execute_access == sa.true())
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/studies_dispatcher/_catalog.py:100]: It seems this branch still does not merge my PR. https://github.com/ITISFoundation/osparc-simcore/pull/6654

File: services/web/server/src/simcore_service_webserver/users/_schemas.py [hunk 32]
```diff
@@ -12,41 +12,41 @@
...
-    state: str | None = Field(description="State, province, canton, ...")
-    postal_code: str | None
-    country: str | None
+    institution: str | None = Field(default=None)
+    phone: str | None = Field(default=None)
+    address: str | None = Field(default=None)
+    city: str | None = Field(default=None)
+    state: str | None = Field(default=None, description="State, province, canton, ...")
+    postal_code: str | None = Field(default=None)
+    country: str | None = Field(default=None)
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/users/_schemas.py:29]: Lets talk about this @giancarloromeo. I had a very specific intention here by making it require. I want to run the idea with you to see if it makes sense

File: services/web/server/src/simcore_service_webserver/utils_aiohttp.py [hunk 33]
```diff
@@ -116,7 +115,7 @@ def create_redirect_to_page_response(
 PageParameters = TypeVar("PageParameters", bound=BaseModel)
 
 
-class NextPage(GenericModel, Generic[PageParameters]):
+class NextPage(BaseModel, Generic[PageParameters]):
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/utils_aiohttp.py:118]: I like how this has changed n pydantic 2! :-)

File: packages/postgres-database/src/simcore_postgres_database/models/products.py [hunk 34]
```diff
@@ -6,11 +6,12 @@
 """
 
 import json
-from typing import Literal, TypedDict
+from typing import Literal
 
 import sqlalchemy as sa
 from sqlalchemy.dialects.postgresql import JSONB
 from sqlalchemy.sql import func
+from typing_extensions import TypedDict
```
[the remark this answers is not in the document]
Comment (giancarloromeo) [reply packages/postgres-database/src/simcore_postgres_database/models/products.py:14]: See the Note: https://docs.pydantic.dev/2.3/usage/types/types_fields/#typeddict

File: packages/common-library/src/common_library/pydantic_validators.py [hunk 35]
```diff
@@ -1,10 +1,35 @@
...
+    # Extract components with defaults if not present
+    sign = match.group("sign") or ""
+    days = match.group("days") or "0"
+    hours = match.group("hours") or "0"
+    minutes = match.group("minutes") or "0"
+    seconds = match.group("seconds")
+    fraction = match.group("fraction") or ""
+
+    # Convert to the format [-][DD]D[,][HH:MM:]SS[.ffffff]
+    return f"{sign}{int(days)}D,{int(hours):02}:{int(minutes):02}:{seconds}{fraction}"
```
[the remark this answers is not in the document]
Comment (giancarloromeo) [reply packages/common-library/src/common_library/pydantic_validators.py:30]: Yes

File: packages/models-library/src/models_library/api_schemas_webserver/clusters.py (hunk 6 above)
Comment (giancarloromeo) [reply packages/models-library/src/models_library/api_schemas_webserver/clusters.py:11]: I didn't change anything here. It was like so in `master` as well.

File: services/web/server/src/simcore_service_webserver/api_keys/_db.py (hunk 17 above)
Comment (giancarloromeo) [reply services/web/server/src/simcore_service_webserver/api_keys/_db.py:82]: ApiKey was not changed.
